### PR TITLE
1319: Rename endpoint call for parent and child subscriptions

### PR DESCRIPTION
**Pull request by tjeerddie**
- rename `parent_subscriptions` endpoint call to `in_use_by`.
- rename `parent_subscriptions` translation to `in_use_by_subscriptions`.
- rename `toggle_hide_child_subscriptions` to `toggle_hide_depends_on_subscriptions`.
- rename `parent_relations` to `in_use_by_relations`.
- rename `children_relations` to `depends_on_relations`.

### CODE REVIEW

File: src/locale/en.ts [hunk 1]
```diff
@@ -364,8 +364,8 @@ I18n.translations.en = {
         notFound: "No Subscription found (e.g. 404)",
         subscription_title: "Subscription",
         child_subscriptions: "Child Subscriptions related to {{product}}",
-        parent_subscriptions: "Child Subscriptions",
-        toggle_hide_child_subscriptions: "Hide terminated child subscriptions",
+        in_use_by_subscriptions: "Child Subscriptions",
```
Comment (tjeerddie) [inline src/locale/en.ts:367]: don't know if we want to change the actual translations

Comment (tjeerddie) [reply src/locale/en.ts:367]: discussed with @wouter1975 and changing it to `Related Subscriptions`

File: src/components/subscriptionDetail/Renderers.tsx [hunk 2]
```diff
@@ -401,11 +401,11 @@ export function RenderSubscriptions({ parentSubscriptions }: { parentSubscriptio
             name={
                 <EuiFlexGroup justifyContent="spaceBetween">
                     <EuiFlexItem grow={false}>
-                        <FormattedMessage id="subscription.parent_subscriptions" />
+                        <FormattedMessage id="subscription.in_use_by_subscriptions" />
                     </EuiFlexItem>
                     <EuiFlexItem>
                         <EuiSwitch
-                            label={<FormattedMessage id="subscription.toggle_hide_child_subscriptions" />}
+                            label={<FormattedMessage id="subscription.toggle_hide_dependent_subscriptions" />}
```
Comment (acidjunk) [inline src/components/subscriptionDetail/Renderers.tsx:408]: Did you miss this one with find/replace?

File: src/utils/types.ts [hunk 3]
```diff
@@ -66,8 +66,8 @@ export interface SubscriptionProcesses {
 
 export interface SubscriptionInstance {
     subscription_id: string;
-    parent_relations: SubscriptionInstanceParentRelation[];
-    children_relations: SubscriptionInstanceParentRelation[];
+    in_use_by_relations: SubscriptionInstanceParentRelation[];
+    depends_relations: SubscriptionInstanceParentRelation[];
```
Comment (acidjunk) [inline src/utils/types.ts:70]: Is see why `dependant/dependent on` could also work (in this place it reads easier I think)

FWIW:
`depends_on`, `used_by` would probably increase readability in code constructs. So without _relations.
But I can see that's it's difficult to make it consistent. We already had several ways to describe: has_children, children relations/parent_subscriptions etc. There are a lot of different post -and suffixes

**Review by acidjunk — APPROVED** [review]
Approved with some nitpicking comments

File: src/utils/types.ts (hunk 3 above)
Comment (tjeerddie) [reply src/utils/types.ts:70]: yes, I do question if they are even used, am not finding it anywhere

**Review by Mark90 — APPROVED** [review]

**Pull request merged**